staging/mausezahn.c

```diff
@@ -676,11 +676,33 @@ int getopts (int argc, char *argv[])
 				(!ipv6_mode && get_ip_range_src(tx.ip_src_txt))
 				)
 			{
-				// name2addr4 accepts a DOTTED DECIMAL ADDRESS or a FQDN:
-				if (ipv6_mode)
+				// name2addr{4,6} accept a valid IP address or a FQDN:
+				if (ipv6_mode) {
 					tx.ip6_src = libnet_name2addr6 (l, tx.ip_src_txt, LIBNET_RESOLVE);
-				else
-					tx.ip_src = libnet_name2addr4 (l, tx.ip_src_txt, LIBNET_RESOLVE);
+					struct in6_addr src_check;
+					if (
+					(strncmp((char*)&tx.ip6_src,(char*)&in6addr_error,sizeof(in6addr_error)) == 0) &&
+					// additional inet_pton() check to cover case where ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff
+					// is specified as source
+					(inet_pton(AF_INET6, tx.ip_src_txt, &src_check) != 1)
+					)
+					{
+						fprintf(stderr, "Failed to set source IPv6 address. Please check if source is set to a valid IPv6 address.\n");
+						return 1;
+					}
+				}
+				else {
+					struct in_addr src_check;
+					if (
+					((tx.ip_src = libnet_name2addr4 (l, tx.ip_src_txt, LIBNET_RESOLVE)) == -1) &&
+					// additional inet_pton() check to cover case where 255.255.255.255 is specified as source
+					(inet_pton(AF_INET, tx.ip_src_txt, &src_check) != 1)
+					)
```

Comment:
Heh, this "if (" expression broken makes C look like a Lisp, a bit hard to read, may be the following would be better:

    ip_src = libnet_name2addr(...)
    if (...) {
    }